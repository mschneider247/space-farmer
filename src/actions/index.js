export const setUser = user => ({
  type: 'SET_USER_NAME',
  user
});

export const setProposal = proposal => ({
  type: 'SET_PROPOSAL',
  proposal
})

export const addProposal = proposal => ({
  type: 'ADD_PROPOSAL',
  proposal
})