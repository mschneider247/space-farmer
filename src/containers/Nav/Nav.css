nav {
  margin-left: 8%;
  margin-bottom: 4%;
  display: flex;
  justify-content: space-between;
  padding: 10px;
}

.div_welcome-logout {
  text-align: center;
  width: 200px;
  margin-right: 8%;
}

.nav_logout-btn {
  width: 200px;
}

button {
  border-radius: 30px;
}