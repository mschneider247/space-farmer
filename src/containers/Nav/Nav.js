import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { setUser } from '../../actions';
import './Nav.css';
import PropTypes from 'prop-types';

export const Nav = (props) => {
  const handleClick = () => {
    props.setUser('')
  }

  const checkLogin = () => {
    if (props.user === ""){
      return (
        <Redirect to='/'></Redirect>
      )
    }
  }

  return (
    <nav>
      {checkLogin()}
      <h1>SPACE <span className="farmer">FARMER</span></h1>
      <div className="div_welcome-logout">
        <p>Welcome! {props.user}</p>
        <button className="nav_logout-btn" onClick={() => handleClick()}>Log Out</button>
      </div>
    </nav>
  )
}

export const mapStateToProps = state => ({
  user: state.setUser
})

export const mapDispatchToProps = dispatch => ({
  setUser: userName => dispatch(setUser(userName))
})

export default connect(mapStateToProps, mapDispatchToProps)(Nav);

Nav.propTypes = {
  user: PropTypes.string,
  setUser: PropTypes.func
}