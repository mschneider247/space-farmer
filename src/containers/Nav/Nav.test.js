import React from 'react';
import { shallow } from 'enzyme';
import { Nav, mapDispatchToProps, mapStateToProps } from './Nav';
import { setUser } from '../../actions';


describe('Nav container', () => {
  it('Nav should match its snapshot', () => {
    const wrapper = shallow(<Nav/>)
    expect(wrapper).toMatchSnapshot();
  });
})

describe('Nav mapDispatchToProps', () => {
  it('Should dispatch setUser with a userName', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = setUser({ id: 1 });
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setUser({ id: 1 });
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});

describe('Nav mapStateToProps', () => {
  it('Should access the setUser key in the Redux store', () => {
    const mockState = {
      setUser: 'Steve',
    };
    const expected = { user: 'Steve' };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});
