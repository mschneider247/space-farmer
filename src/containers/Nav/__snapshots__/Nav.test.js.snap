// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match its snapshot 1`] = `ShallowWrapper {}`;
